doc/classes/@GlobalScope.xml

```diff
@@ -2141,135 +2141,36 @@
-		</constant>
-		<constant name="JOY_SONY_START" value="6" enum="JoyButtonList">
-			Sony DualShock controller start button maps to SDL start button.
-		</constant>
-		<constant name="JOY_SONY_PS" value="5" enum="JoyButtonList">
-			Sony DualShock controller PS button maps to SDL guide button.
-		</constant>
-		<constant name="JOY_SONY_L1" value="9" enum="JoyButtonList">
-			Sony DualShock controller L1 button maps to SDL left shoulder button.
-		</constant>
-		<constant name="JOY_SONY_R1" value="10" enum="JoyButtonList">
-			Sony DualShock controller R1 button maps to SDL right shoulder button.
-		</constant>
-		<constant name="JOY_SONY_L3" value="7" enum="JoyButtonList">
-			Sony DualShock controller L3 button maps to SDL left stick button.
-		</constant>
-		<constant name="JOY_SONY_R3" value="8" enum="JoyButtonList">
-			Sony DualShock controller R3 button maps to SDL right stick button.
-		</constant>
-		<constant name="JOY_XBOX_A" value="0" enum="JoyButtonList">
-			Xbox game controller A button maps to SDL button A.
-		</constant>
-		<constant name="JOY_XBOX_B" value="1" enum="JoyButtonList">
-			Xbox game controller B button maps to SDL button B.
-		</constant>
-		<constant name="JOY_XBOX_X" value="2" enum="JoyButtonList">
-			Xbox game controller X button maps to SDL button X.
-		</constant>
-		<constant name="JOY_XBOX_Y" value="3" enum="JoyButtonList">
-			Xbox game controller Y button maps to SDL button Y.
-		</constant>
-		<constant name="JOY_XBOX_BACK" value="4" enum="JoyButtonList">
-			Xbox game controller back button maps to SDL back button.
-		</constant>
-		<constant name="JOY_XBOX_START" value="6" enum="JoyButtonList">
-			Xbox game controller start button maps to SDL start button.
-		</constant>
-		<constant name="JOY_XBOX_HOME" value="5" enum="JoyButtonList">
-			Xbox game controller home button maps to SDL guide button.
-		</constant>
-		<constant name="JOY_XBOX_LS" value="7" enum="JoyButtonList">
-			Xbox game controller left stick button maps to SDL left stick button.
-		</constant>
-		<constant name="JOY_XBOX_RS" value="8" enum="JoyButtonList">
-			Xbox game controller right stick button maps to SDL right stick button.
-		</constant>
-		<constant name="JOY_XBOX_LB" value="9" enum="JoyButtonList">
-			Xbox game controller left bumper button maps to SDL left shoulder button.
-		</constant>
-		<constant name="JOY_XBOX_RB" value="10" enum="JoyButtonList">
-			Xbox game controller right bumper button maps to SDL right shoulder button.
-		</constant>
 		<constant name="JOY_BUTTON_MAX" value="36" enum="JoyButtonList">
 			The maximum number of game controller buttons.
 		</constant>
-		<constant name="JOY_INVALID_AXIS" value="-1" enum="JoyAxisList">
+		<constant name="JOY_AXIS_INVALID" value="-1" enum="JoyAxisList">
 			An invalid game controller axis.
 		</constant>
 		<constant name="JOY_AXIS_LEFT_X" value="0" enum="JoyAxisList">
-			Game controller left joystick x-axis.
+			Game controller SDL left joystick x-axis.
```

Comment:
Is "(or generic joystick index 0)" necessary? If users want "index 0" and don't care about what it means, they can just pass the number 0 instead of `JOY_AXIS_LEFT_X`.

Reply:
"joystick index 0" is now about the enum value (that's a coincidence here), it's to match this: https://github.com/godotengine/godot/pull/43591/files#diff-f8cadb6781bc795c8371be8b6dced5972f5e85de3e76839491baa8edba9f78a4R56

Arbitrary joysticks are numbered 0 to 4.

See e.g. https://www.elgiganten.dk/product/gaming/gaming-controllere-joystick-rat-og-pedaler/THRTHOTASX/thrustmaster-t-flight-hotas-x-joystick, I don't own it but I think the main flight stick might be index 0, and the thrust on the left might be index 1.

The description states that it has 5 axes so there's likely one like up/down on the flight stick which might have index 3.

Reply:
To clarify, the parenthesis provides an alternative to "on SDL game controller", not to the whole sentence. Suggestions welcome for better phrasing.

Reply:
I've gone with
"Game controller left joystick x-axis. Corresponds to axis 0."
"Game controller left joystick y-axis. Corresponds to axis 1."
"Game controller right joystick x-axis. Corresponds to axis 2." etc.

Reply:
Well now it's incorrect IMO, and indeed just duplicates the value of the enum itself. It doesn't match what you have in InputMapEditor.

My understanding is that there's two representation:

- On a SDL gamepad, you have two sticks (left and right). The left stick has X axis (index 0) and Y axis (index 1). The right stick has X axis (index 2) and Y axis (index 3).
- On a generic joystick, **if the input map editor is correct**, you can have any number of **sticks** (well up to 4 for us), each with X and Y axes. So there index 0 is the X axis of the stick no. 0, 1 is the Y axis of the stick no. 0, 2 is the X axis of the stick no. 1, 3 is the Y axis of the stick no. 1, 4 is the X axis of the stick no. 2, etc.

Reply:
Each joystick has two axes, and each axis has two directions. What's confusing is that there is a left and a right joystick, which each have a left and a right direction.

The SDL game controller mappings support 6 axes: 2 joysticks (left and right) with 2 axes each (X and Y) with 2 directions each (+ve and -ve) and two triggers, each with its own axis and only positive values. These six axes are what the `enum` corresponds to.

On the other hand, Godot provides support for 5 joysticks, 10 axes and 20 directions. The map editor provides those 20 directions as options: and prepends each of them with the axis number and a `+` or `-` for the direction:

![Screenshot_2020-11-19_14-24-05](https://user-images.githubusercontent.com/9253928/99678818-237e0880-2a73-11eb-83ec-334b4c45e6ed.png)

The way I see it, we have two options:
1. Drop the "Corresponds to axis x":
- JOY_AXIS_LEFT_X: Game controller left joystick x-axis.
- JOY_AXIS_LEFT_Y:  Game controller left joystick y-axis.
- JOY_AXIS_RIGHT_X: Game controller right joystick x-axis.
- JOY_AXIS_RIGHT_Y: Game controller right joystick y-axis.
- JOY_AXIS_TRIGGER_LEFT: Game controller left trigger axis.
- JOY_AXIS_TRIGGER_RIGHT: Game controller right trigger axis.
2. Change them to:
- JOY_AXIS_LEFT_X: Game controller left joystick x-axis. Corresponds to Joystick 0 X.
- JOY_AXIS_LEFT_Y:  Game controller left joystick y-axis. Corresponds to Joystick 0 Y.
- JOY_AXIS_RIGHT_X: Game controller right joystick x-axis. Corresponds to Joystick 1 X.
- JOY_AXIS_RIGHT_Y: Game controller right joystick y-axis. Corresponds to Joystick 1 Y.
- JOY_AXIS_TRIGGER_LEFT: Game controller left trigger axis. Corresponds to Joystick 2 Right.
- JOY_AXIS_TRIGGER_RIGHT: Game controller right trigger axis. Corresponds to Joystick 2 Down.

I prefer the first option, drop the "Corresponds to axis x" i.e. the way it is currently. The second option, although correct, adds confusion, because the triggers are positive only.

Reply:
I agree, let's go with option 1. then.

Reply:
Done.